src/WebJobs.Script/PropertyHelper.cs

```diff
@@ -0,0 +1,513 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Reflection;
+
+namespace Microsoft.Azure.WebJobs.Script
+{
+    // Original code here: https://github.com/aspnet/Common/blob/dev/src/Microsoft.Extensions.PropertyHelper.Sources/PropertyHelper.cs
```

Comment:
This is a common helper class floating around in a lot of MSFT projects. We have one in WebJobs SDK as well, but it is an incomplete early version of this (it didn't have setter support).